pkg/cri/server/container_create_linux.go

```diff
@@ -270,9 +270,37 @@ func (c *criService) containerSpec(
 		specOpts = append(specOpts, customopts.WithAnnotation(pKey, pValue))
 	}
 
+	specOpts = append(specOpts, customopts.WithPodNamespaces(securityContext, sandboxPid))
+
+	if securityContext.GetNamespaceOptions().GetPid() == runtime.NamespaceMode_TARGET {
+		targetContainerID := securityContext.GetNamespaceOptions().TargetId
+		targetContainer, err := c.containerStore.Get(targetContainerID)
+		if err != nil {
+			return nil, errors.Wrapf(err, "failed to find target container %q", targetContainerID)
+		}
+
+		if targetContainer.Metadata.SandboxID != sandboxID {
+			return nil, errors.Errorf("target container belongs to a different sandbox")
+		}
+
+		status := targetContainer.Status.Get()
+		if state := status.State(); state != runtime.ContainerState_CONTAINER_RUNNING {
+			return nil, errors.Errorf("target container is in state %s", state)
+		}
+
+		targetPid := status.Pid
+
+		// Overwrite the PID namespace with the namespace of the target container.
+		specOpts = append(specOpts, oci.WithLinuxNamespace(
```

Comment:
instead of setting it then overwriting it ... let's add an additional parameter to withpodnamespaces for the targetpid,.. then inside withpodnamespaces name the first pid sandboxpid... and the new second one .. targetpid.. then you ca keep the with call below... and set var targetpid equal to sanboxpid and overwrite it in this runtime.NamespaceMode_TARGET code block.. :-)

Reply:
nit.. maybe move the validation checks to a validation call.. then you would only need to unit test the validation call.

Reply:
Agree for both. :)

Reply:
+1